assignments/week_1/exercise_1_2/exercise_1_2.md

```diff
@@ -0,0 +1,231 @@
+
+int main()
+{
+    // Part 2: Palindrome Check
+
+    // Input string from command-line
+    std::string str;
+    std::cout << "Ender String:" << std::endl;
+    std::cin >> str;
+
+    // Palindrome Check
+    // WRITE YOUR PALINDROME CHECK HERE
+
+    return 0;
+}
+```
+
+1. Create variables for the string's leftmost and rightmost indices (`left` and `right`)
+<details>
+  <summary>Answer</summary>
+
+  ```c++
+    int left = 0;
+    int right = str.length() - 1;
+  ```
+
+</details>
+
+2. Create a variable (`isPalindrome`) that will store the result of the palindrome check
+<details>
+  <summary>Answer</summary>
+
+  ```c++
+    bool isPalindrome = true;
+  ```
+
+</details>
+
+3. Create a loop which continues as long as `left` is less than `right`
+<details>
+  <summary>Answer</summary>
+
+  ```c++
+    while (left < right)
+    {
+        // MORE CODE
+    }
+  ```
+
+</details>
+
+4. In the loop, check if the elements at `left` and `right` in the string are different
+<details>
+  <summary>Answer</summary>
+
+  ```c++
+    while (left < right)
+    {
+        if (str[left] != str[right])
+        {
+            // MORE CODE
+        }
+    }
+  ```
+
+</details>
+
+5. If the elements are different, then the string should be marked are not a palindrome and you should leave the loop
+<details>
+  <summary>Answer</summary>
+
+  ```c++
+    while (h > l)
+    {
+        if (str[l] != str[h])
+        {
+            isPalindrome = false;
+            break;
+        }
+    }
+  ```
+
+</details>
+
+6. In the loop, increment `left` and decrement `right` so that `left` will iterate forwards over the first half and `right` will iterate backwards over the second half
+<details>
+  <summary>Answer</summary>
+
+  ```c++
+    while (left < right)
+    {
+        if (str[left] != str[right])
+        {
+            isPalindrome = false;
+            break;
+        }
+        left++;
+        right--;
+    }
+  ```
+
+</details>
+
+7. Print out the results (ex. <str\> is (NOT) a palindrome)
+<details>
+  <summary>Answer</summary>
+
+  ```c++
+    if (isPalindrome)
+        std::cout << str << " is a palindrome" << std::endl;
+    else
+        std::cout << str << " is NOT a palindrome" << std::endl;
+  ```
+
+</details>
+
+Now run the program and test multiple strings to check your program works!
+
+# Review
+Great Job! You successfully manipulated a vector and string in this exercise. Although, these intructions explains one method to do vector reversal and palindrome
+checking, there exists much easier ways to do these operations using C++ standard library functions. For example, `std::reverse(list.begin(), list.end());` would
+reversal the vector in just one line without any hassle! Also, `str == string(str.rbegin(), str.rend())` would check if `str` is a palindrome in one line as well.
+The standard library is powerful tool for writing C++ code and as you progress through future weeks' exercises you become more and more familiar with it!
```

Comment:
Suggestion: (can't use the normal suggestion feature because my suggested text includes code blocks.

Great Job! You successfully manipulated a vector and string in this exercise. Although, these instructions explain one method to do vector reversal and palindrome checking, there are much easier ways to do these operations using C++ standard library functions. 

For example, [`std::reverse()`](https://en.cppreference.com/w/cpp/algorithm/reverse) lets you easily reverse any standard container.

```
std::reverse(list.begin(), list.end());
```

For palindrome checking, there are multiple one-line solutions using the standard library features. Standard containers (including `std::string`) have reverse iterators that let you iterate over the container in reverse order. So the palindrome check could be written like this:

```
bool is_palindrome = ( str == string(str.rbegin(), str.rend()) );
```

Or even in a way that's generic for any standard container:

```
bool is_palindrome = std::equal(str.begin(), str.begin() + str.size()/2, str.rbegin());
```

Checkout [`std::string::rbegin()`](https://en.cppreference.com/w/cpp/string/basic_string/rbegin) and [`std::string::rend()`](https://en.cppreference.com/w/cpp/string/basic_string/rend) for documentation.

The standard library is a powerful tool for writing C++ code. As you progress through future weeks' exercises, you'll become more and more familiar with it!